bugtool/cmd/root.go

```diff
@@ -331,7 +339,7 @@ func runAll(commands []string, cmdDir string, k8sPods []string) {
 			// When we are done we return the thing we took from
 			// the semaphore, so another goroutine can get it
 			defer func() { semaphore <- true }()
-			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown)
+			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown, nil)
```

Comment:
In your current code:

1. You have a specific case for the `ip xfrm state` command (above, in your earlier chunk), where you call `writeCmdToFile()` with the dedicated callback. This writes the output to the expected file for that command.
2. Then, here in this chunk, you call `writeCmdToFile()` for _all_ commands, _including_ `ip xfrm state`, and write to the corresponding output files. So you're actually overwriting your processed output containing the hashed keys with the regular output produced by `writeCmdToFile()` without filtering callback.

I think this is the reason why the files in your archive contain keys that are unchanged. What you want instead is to process all commands at the same place, in the goroutine, with just a check to see if the callback should be passed. Like this:

```suggestion
		if strings.Contains(cmd, "xfrm") && strings.Contains(cmd, "state") {
			// Special case for hashing the keys for `ip xfrm state`
			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown, hashIPseckeys)
		} else {
			// Normal case for all other commands
			writeCmdToFile(cmdDir, cmd, k8sPods, enableMarkdown, nil)
		}
```

Reply:
I just understood, I was overwriting my files. Thankyou !